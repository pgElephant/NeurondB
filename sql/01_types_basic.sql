-- Test basic type creation and operations
CREATE EXTENSION neurondb;

-- Test vector type
SELECT '[1.0, 2.0, 3.0]'::vector;
SELECT '[1.5, 2.5, 3.5, 4.5]'::vector;
SELECT vector_dims('[1.0, 2.0, 3.0, 4.0, 5.0]'::vector);

-- Test vectorp (packed)
SELECT vectorp_in('[1.0, 2.0, 3.0]')::text;
SELECT vectorp_dims(vectorp_in('[1.0, 2.0, 3.0, 4.0]'));

-- Test vecmap (sparse)
SELECT vecmap_in('{dim:10, nnz:3, indices:[0,3,7], values:[1.5,2.3,0.8]}')::text;

-- Test vgraph
SELECT vgraph_in('{nodes:5, edges:[[0,1],[1,2],[2,3],[3,4]]}')::text;

-- Test rtext
SELECT rtext_in('sample text for retrieval')::text;
