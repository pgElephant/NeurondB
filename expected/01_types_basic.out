-- Test basic type creation and operations
CREATE EXTENSION neurondb;
ERROR:  could not open extension control file "/opt/homebrew/share/postgresql@14/extension/neurondb.control": No such file or directory
-- Test vector type
SELECT '[1.0, 2.0, 3.0]'::vector;
ERROR:  type "vector" does not exist
LINE 1: SELECT '[1.0, 2.0, 3.0]'::vector;
                                  ^
SELECT '[1.5, 2.5, 3.5, 4.5]'::vector;
ERROR:  type "vector" does not exist
LINE 1: SELECT '[1.5, 2.5, 3.5, 4.5]'::vector;
                                       ^
SELECT vector_dims('[1.0, 2.0, 3.0, 4.0, 5.0]'::vector);
ERROR:  type "vector" does not exist
LINE 1: SELECT vector_dims('[1.0, 2.0, 3.0, 4.0, 5.0]'::vector);
                                                        ^
-- Test vectorp (packed)
SELECT vectorp_in('[1.0, 2.0, 3.0]')::text;
ERROR:  function vectorp_in(unknown) does not exist
LINE 1: SELECT vectorp_in('[1.0, 2.0, 3.0]')::text;
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
SELECT vectorp_dims(vectorp_in('[1.0, 2.0, 3.0, 4.0]'));
ERROR:  function vectorp_in(unknown) does not exist
LINE 1: SELECT vectorp_dims(vectorp_in('[1.0, 2.0, 3.0, 4.0]'));
                            ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- Test vecmap (sparse)
SELECT vecmap_in('{dim:10, nnz:3, indices:[0,3,7], values:[1.5,2.3,0.8]}')::text;
ERROR:  function vecmap_in(unknown) does not exist
LINE 1: SELECT vecmap_in('{dim:10, nnz:3, indices:[0,3,7], values:[1...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- Test vgraph
SELECT vgraph_in('{nodes:5, edges:[[0,1],[1,2],[2,3],[3,4]]}')::text;
ERROR:  function vgraph_in(unknown) does not exist
LINE 1: SELECT vgraph_in('{nodes:5, edges:[[0,1],[1,2],[2,3],[3,4]]}...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- Test rtext
SELECT rtext_in('sample text for retrieval')::text;
ERROR:  function rtext_in(unknown) does not exist
LINE 1: SELECT rtext_in('sample text for retrieval')::text;
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
